app/models.py

```diff
@@ -422,7 +427,7 @@ def serialize(self, data=None):
         }, **(data or {}))
 
         if self.framework_agreements:
```

Comment:
Should we still check `self.framework_agreements` here or should we now check `self.current_framework_agreement`?